goose.go

```diff
@@ -1928,17 +1962,10 @@ func (ctx Ctx) returnType(results *ast.FieldList) coq.Type {
 		return coq.TypeIdent("unitT")
 	}
 	rs := results.List
-	for _, r := range rs {
-		if len(r.Names) > 0 {
-			ctx.unsupported(r, "named returned value")
```

Comment:
I was trying to retrieve the type of each method defined in the interface, and it seems that this loop prevents access to the name of the type. Maybe there's a way around this? 

For instance, within this interface:
```
type greeting interface {
   intro() string
}
```
I want to retrieve `string`. The loop rejects named returned values so I'm unable to retrieve `string` for `intro()`. I may be misunderstanding the implementation, though. 

I'll add the negative test to prevent the failure from happening in future translations.

Reply:
In that example `string` isn't named. The representation is that `rs` is a list with one element (for the one return value), and it has `r.Names` of `nil` (because it doesn't have a name).

I haven't looked specifically at types in interfaces so they might be represented differently, but note that you can also write `intro() (s string)` and use a named return value.